fsmonitor-settings.c

```diff
@@ -78,12 +101,16 @@ static void lookup_fsmonitor_settings(struct repository *r)
 		return;
 
 	case 1: /* config value was unset */
+		if (check_deprecated_builtin_config(r))
+			return;
+
 		const_str = getenv("GIT_TEST_FSMONITOR");
```

Comment:
Just like `core.fsmonitor` overrides `GIT_TEST_FSMONITOR`, we now also let `core.useBuiltinFSMonitor` override it. Good!